internal/controller/pkg/revision/runtime.go

```diff
@@ -159,6 +159,7 @@ func (b *RuntimeManifestBuilder) ServiceAccount(overrides ...ServiceAccountOverr
 
 		// Overrides that we are opinionated about.
 		ServiceAccountWithNamespace(b.namespace),
+		ServiceAccountWithLabels(b.podSelectors()),
```

Comment:
Perhaps we should rename `podSelectors` if we're now using it to do more than select pods.